gotests.go

```diff
@@ -18,13 +18,14 @@ import (
 
 // Options provides custom filters and parameters for generating tests.
 type Options struct {
-	Only        *regexp.Regexp        // Includes only functions that match.
-	Exclude     *regexp.Regexp        // Excludes functions that match.
-	Exported    bool                  // Include only exported methods
-	PrintInputs bool                  // Print function parameters in error messages
-	Subtests    bool                  // Print tests using Go 1.7 subtests
-	Importer    func() types.Importer // A custom importer.
-	TemplateDir string                // Path to custom template set
+	Only          *regexp.Regexp         // Includes only functions that match.
+	Exclude       *regexp.Regexp         // Excludes functions that match.
+	Exported      bool                   // Include only exported methods
+	PrintInputs   bool                   // Print function parameters in error messages
+	Subtests      bool                   // Print tests using Go 1.7 subtests
+	Importer      func() types.Importer  // A custom importer.
+	TemplateDir   string                 // Path to custom template set
+	ExternalParas map[string]interface{} // Custom external parameters
```

Comment:
Please create two options: `TemplateParams` for the map and `TemplateParamsPath` for the string path, to be processed downstream. This will make testing easier.